plugins/github-octokit/src/api/github-credentials-api.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { GitHubIntegrationConfig } from '@backstage/integration';
+import { Entity } from '@backstage/catalog-model';
+
+import { OwnerRepo } from '../utils/types';
+
+export type IntegrationInfo = { baseUrl: string; host: string };
+export type CredentialsInfo = IntegrationInfo & { token: string };
+
+export type GithubCredentialsApi = {
```

Comment:
There's significant overlap between this and the existing [ScmAuthApi](https://github.com/backstage/backstage/blob/6471765816c7c13d718cb636bada2662bde7a88d/packages/integration-react/src/api/ScmAuthApi.ts#L85) and [ScmIntegrationsApi](https://github.com/backstage/backstage/blob/6471765816c7c13d718cb636bada2662bde7a88d/packages/integration-react/src/api/ScmIntegrationsApi.ts#L45). I'm wondering where this would end up if we utilized those instead? Perhaps what's in this package could potentially be additions to `integrations-react` instead?